woocommerce/woocommerce-gutenberg-products-block
# introduce SlotFill for Sidebar

This PR Implements a Slot that is rendered in both Cart and Checkout, just after `Totals` section.

This slot wrap each fill in an error boundary, so that if a plugin code breaks, it would not affect the block or any other extension hooking into that slot.

### Implmentation
```jsx
import { ExperimentalOrderMeta } from '@woocommerce/blocks-checkout';

const MyApp = () => {
  return (
    <ExperimentalOrderMeta>My stuff here</ExperimentalOrderMeta />
  );
}

registerPlugin( 'my-app', { render: MyApp } );
```

Closes #3246

### Testing instructions 
- The easiest way to test this is to checkout this branch 3940-gh-woocommerce/woocommerce-subscriptions
- In Cart or Checkout, you should see `This is my default export`
- Now to test error boundary, in the above plugin `assets/src/js/index.js` or anywhere in our plugin, add this code
```jsx
const myRender = () => {
	return (
		<ExperimentalOrderMeta>
			<div className="block-div-class">This is my other export</div>
		</ExperimentalOrderMeta>
	);
};

registerPlugin( 'woocommerce-two-three', {
	render: myRender,
} );
```
- Modify the original code so it throws a **runtime** error.
```jsx
const Bug = () => {
	throw Error( 'bug' );
};
const render = () => {
	const [ showBug, setShowBug ] = useState( false );
	return (
		<ExperimentalOrderMeta>
			<div
				className="block-div-class"
				onClick={ () => setShowBug( true ) }
			>
				This is my default export
			</div>
			{ showBug && <Byg /> }
		</ExperimentalOrderMeta>
	);
};
```
- On cart or Checkout, click on the text to trigger the error.
- Only that component is removed from the page, the rest should still be fine
